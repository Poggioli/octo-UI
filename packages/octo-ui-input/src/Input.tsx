import { forwardRef } from "@octo-ui/core";
import { StyledInput } from "./styles";
import { InputProps } from "./types";

const Input = forwardRef<typeof StyledInput, InputProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledInput {...props} ref={forwardedRef} />;
  }
);
Input.toString = () => `.${StyledInput.className}`;

export { StyledInput };
export { Input };

