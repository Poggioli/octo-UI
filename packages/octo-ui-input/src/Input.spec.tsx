import { render } from "@testing-library/react";
import { Input } from "./Input";

describe("<Input />", () => {
  it(`GIVEN a <Input />
      WHEN render is called
      THEN should render a <Input />`, () => {
    const rendered = render(<Input />);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Input />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Input.toString()).toBe(".octo-UI-c-eQegQU");
  });
});
