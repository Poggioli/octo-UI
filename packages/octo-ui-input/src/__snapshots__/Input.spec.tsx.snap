// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> GIVEN a <Input />
      WHEN render is called
      THEN should render a <Input /> 1`] = `
<div>
  <input
    class="octo-UI-c-eQegQU octo-UI-c-eQegQU-cQBoFB-size-standard octo-UI-c-eQegQU-dhMCoC-color-unstyled"
  />
</div>
`;
